### DISCUSSION TITLE
Pylint complains about accessing cached properties

### DISCUSSION THREAD

**Discussion by hofrob** [aio-libs/yarl#1330] [General]
Hello everyone!

Since version 1.14.0 `pylint` complains about accessing `URL.host` and other cached properties. I'm not sure what's the cause of this error so I wanted to ask here. There's an example project here: https://github.com/hofrob/yarl-pylint-cached-prop

I was not able to reproduce this using the `uv` parameter `--with`, so you'd have to edit the dependencies manually.

Can anyone confirm or do you have any pointers what went wrong? And in which project? :grinning: 

Thank you!
hofrob

**Comment by webknjaz** [comment]
It's best to integrate some sort of CI in such examples to demonstrate what the output is and what the invocation looks like. And avoid fancy tools, sticking to pip et al: https://sscce.org.

The version you mentioned migrated the vendored internal cached property implementation into a shared library `propcache`. Perhaps, that's what confuses `pylint`.

It's implemented in a C-extension so I wouldn't be surprised if it needed https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#extension-pkg-allow-list.

Do you have all the deps installed in the pylint env, by the way?

Have you tried editing your settings to include `propcache.api.under_cached_property` per https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#property-classes ?

cc @bdraco

**Comment by bdraco** [comment]
It might be this issue https://github.com/pylint-dev/pylint/issues/3484
